src/utils/getTestFrameworkName.ts

```diff
@@ -2,12 +2,12 @@
 import type { PackageJSON } from "@nodesecure/npm-types";
 
 // CONSTANTS
-const kNodeTestRunnerRegex = /(?:node\s+--test|tsx)/;
+const kNodeTestRunnerRegex = /node\s+.*--test|tsx/;
 
 export function getTestFrameworkName(packageJson: PackageJSON): string {
   const { devDependencies: deps = {}, scripts = {} } = packageJson;
 
-  if (kNodeTestRunnerRegex.test(scripts.test)) {
+  if (kNodeTestRunnerRegex.test(scripts.test) || kNodeTestRunnerRegex.test(scripts["test-only"]) || "tsx" in deps) {
```

Comment:
`|| "tsx" in deps` is wrong.
`tsx` is a library used to run TS without building first. Here you'll detect `node:test` for any TS project that use `tsx` even if the projet don't use `node:test`. (same problem with the regexp)
```suggestion
  if (kNodeTestRunnerRegex.test(scripts.test) || kNodeTestRunnerRegex.test(scripts["test-only"])) {
```